vsintegration/src/FSharp.Editor/Formatting/IndentationService.fs

```diff
@@ -93,18 +94,11 @@ type internal FSharpIndentationService
                     lastIndent
         }
 
-    interface IFSharpSynchronousIndentationService with
-        member this.GetDesiredIndentation(document: Document, lineNumber: int, cancellationToken: CancellationToken): Nullable<FSharpIndentationResult> =
-            async {
-                let! cancellationToken = Async.CancellationToken
-                let! sourceText = document.GetTextAsync(cancellationToken) |> Async.AwaitTask
-                let! options = document.GetOptionsAsync(cancellationToken) |> Async.AwaitTask
-                let tabSize = options.GetOption<int>(FormattingOptions.TabSize, FSharpConstants.FSharpLanguageName)
-                let indentStyle = options.GetOption(FormattingOptions.SmartIndent, FSharpConstants.FSharpLanguageName)
-                let parsingOptions = document.GetFSharpQuickParsingOptions()
-                let indent = FSharpIndentationService.GetDesiredIndentation(document.Id, sourceText, document.FilePath, lineNumber, tabSize, indentStyle, parsingOptions)
-                return
-                    match indent with
-                    | None -> Nullable()
-                    | Some(indentation) -> Nullable<FSharpIndentationResult>(FSharpIndentationResult(sourceText.Lines.[lineNumber].Start, indentation))
-            } |> (fun c -> Async.RunImmediateExceptOnUI(c,cancellationToken=cancellationToken))
+    interface IFSharpIndentationService with
+        member this.GetDesiredIndentation(services: HostLanguageServices, text: SourceText, documentId: DocumentId, path: string, lineNumber: int, tabSize: int, indentStyle: FormattingOptions.IndentStyle): Nullable<FSharpIndentationResult> =
```

Comment:
Is there any case when F# document would have `null` path?

Reply:
Not clear to me what the tokenizer is doing with `path`. Seems like it's used for a couple of purposes:
1) determine whether document is a script (`.fsx` extension)
2) error location reporting?

Previous code passed `document.FileName` to the tokenizer where it is used to check `.fsx` extension (but this may be null). `TryGetQuickParsingOptionsForEditingDocumentOrProject` used `document.Name` to check `.fsx` extension.